### ISSUE TITLE
False negative "missing displayName"

### ISSUE THREAD

**Issue by jasonkuhrt** [yannickcr/eslint-plugin-react#418]
I believe this is a bug in React ESLint plugin.

This code:

``` js
const LoadingBar = ({ progress }) => (
  <Motion style={{ x: spring(progress) }}>{({ x }) =>
    <div className="LoadingBar">
      <div className="LoadingBar-view">
        <div
          className="LoadingBar-bar"
          style={{
            width: `${x}%`,
          }}
        />
      </div>
      <div className="LoadingBar-label">
        {`${toInteger(x)}%`}
      </div>
    </div>
  }</Motion>
)
```

causes:

<img width="640" alt="screen shot 2016-01-29 at 3 18 07 pm" src="https://cloud.githubusercontent.com/assets/284476/12687057/85f753f4-c69b-11e5-8655-5e73993fc326.png">

with package versions:

```
    "eslint": "^1.10.3",
    "eslint-plugin-react": "^3.16.1"
```

**Comment by lencioni** [comment]
Possibly. I think there might be some overlap here with #412.

**Comment by kamek-pf** [comment]
Yup, stateless components are causing warnings for me too.

**Comment by ljharb** [comment]
@jasonkuhrt if you try with the latest version of this plugin, is this still an issue? I'm happy to reopen if so.
